lib/datadog/tracing/trace_operation.rb

```diff
@@ -343,6 +343,8 @@ def subscribe_deactivate_trace(&block)
       def activate_span!(span_op)
         parent = @active_span
 
+        span_op.__send__(:top_level=, true) if parent.nil? || (span_op.service && parent.service != span_op.service)
```

Comment:
`span_op.service` can change until the span is `span_op.finish`ed, this means that `top_level=` would also change.
Can we move this logic to `deactivate_span!` instead?

Reply:
Nice catch! I wrote a failing spec for the scenario that service could be overwritten within the block!

https://github.com/DataDog/dd-trace-rb/pull/2080/commits/9b75906306e8478c98cc084481d92d39ececcd1b